tor/core/admin_commands.py

```diff
@@ -40,13 +41,73 @@ def process_override(reply, config):
     if 'done' in parents_parent.body.lower():
         logging.info(
             'Starting validation override for post {}'
-            ', approved by {}'.format(parents_parent.fullname, reply.author.name)
+            ', approved by {}'.format(
+                parents_parent.fullname,
+                reply.author.name
+            )
         )
         process_done(
             parents_parent, config, override=True
         )
 
 
+def process_blacklist(reply, config):
+    """
+    This is used to basically "shadow-ban" people from the bot.
+    Format is:
+    Subject: !blacklist
+    body: <username1>\n<username2>...
+    :param reply: the comment reply object from the inbox
+    :param config: the global config object
+    :return: None
+    """
+
+    if not from_moderator(reply, config):
+        reply.reply(_(random.choice(config.no_gifs)))
+        logging.info(
+            '{} just tried to blacklist. Get your own bot!'
+            ''.format(reply.author.name)
+        )
+        return
+
+    usernames = reply.body.split('\n')
```

Comment:
I _think_ reddit returns windows-style newlines with a carriage return. Might be worth investigating.

Reply:
I'll check in a bit

Reply:
To make it more agnostic, why not use `str.readlines()`? It should make it agnostic to line endings with `\r\n`, `\n`, or whatever else there might be.